src/specific/s_input.h

```diff
@@ -2,22 +2,48 @@
 
 #include "global/types.h"
 
+#include <SDL2/SDL_gamecontroller.h>
 #include <stdbool.h>
 #include <stdint.h>
 
 void S_Input_Init(void);
 void S_Input_Shutdown(void);
 
-INPUT_STATE S_Input_GetCurrentState(INPUT_LAYOUT layout_num);
+INPUT_STATE S_Input_GetCurrentState(
+    INPUT_LAYOUT layout_num, INPUT_LAYOUT cntlr_layout_num);
 
 INPUT_SCANCODE S_Input_GetAssignedScancode(
     INPUT_LAYOUT layout_num, INPUT_ROLE role);
 
+int16_t S_Input_GetUniqueBind(INPUT_LAYOUT layout_num, INPUT_ROLE role);
+
+int16_t S_Input_GetAssignedButtonType(INPUT_LAYOUT layout_num, INPUT_ROLE role);
+
+int16_t S_Input_GetAssignedBind(INPUT_LAYOUT layout_num, INPUT_ROLE role);
+
+int16_t S_Input_GetAssignedAxisDir(INPUT_LAYOUT layout_num, INPUT_ROLE role);
+
 void S_Input_AssignScancode(
     INPUT_LAYOUT layout_num, INPUT_ROLE role, INPUT_SCANCODE scancode);
 
-bool S_Input_ReadAndAssignKey(INPUT_LAYOUT layout_num, INPUT_ROLE role);
+void S_Input_AssignButton(
+    INPUT_LAYOUT layout_num, INPUT_ROLE role, SDL_GameControllerButton button);
+
+void S_Input_AssignAxis(
+    INPUT_LAYOUT layout_num, INPUT_ROLE role, SDL_GameControllerAxis axis,
+    int16_t axis_dir);
 
-const char *S_Input_GetKeyName(INPUT_LAYOUT layout_num, INPUT_ROLE role);
+void S_Input_ResetControllerToDefault(INPUT_LAYOUT layout_num);
+
+bool S_Input_ReadAndAssignKey(
+    CONTROL_MODE mode, INPUT_LAYOUT layout_num, INPUT_ROLE role);
+
+const char *S_Input_GetKeyName(
+    CONTROL_MODE mode, INPUT_LAYOUT layout_num, INPUT_ROLE role);
+
+const char *S_Input_GetButtonNameFromString(
+    INPUT_LAYOUT layout_num, const char *btn_name);
```

Comment:
This still uses `btn_name`

Reply:
Should I rename `btn` to `button` everywhere like I did `ax`s to `axis`? I originally used `btn` because names like `CONTROL_RESET_DEFAULTS_BTN` and `GS_CONTROL_RESET_DEFAULTS_BTN` are very long already. There are a few remaining names I can change if we want.

Reply:
No just standalone `btn`  → `button`.